src/tools/tidy/src/main.rs

```diff
@@ -171,7 +170,17 @@ fn main() {
         };
         check!(unstable_book, &src_path, collected);
 
-        check!(ext_tool_checks, &root_path, &output_directory, bless, extra_checks, pos_args);
+        check!(
+            ext_tool_checks,
```

Comment:
Why is this being added to a single check module? AFAICS `ext_tools` is running external programs serially, so the wall-time of all those tools adds up. The `check!` macro exists to run things in threads so they can run in parallel to cut down on walltime.

Reply:
It's in `ext_tools` because that's where `--extra-checks` is parsed.  I definitely agree that the parsing logic and stuff could use a big upgrade, and that tools should be run concurrently, but I think that can be done within ext_tools.  It doesn't even technically need threads, just to spawn processes then wait for them all simultaneously.  I could get working on a system for that if you want.  It would probably involve making a struct or trait to represent an extra-check instead of having it be ad-hoc.  This would also be a good opportunity to refactor the parsing.

Reply:
Well, the macro also limits concurrency to obey `-j`, so having thread per check is the easiest way to get that.

Reply:
So, should I inline every `--extra-check` as its own call to `check!`, probably doubling the size of `main()` in the process, or do you have a better idea?

Reply:
Well, before doing more work I think it'd make sense to check how long it takes to run all the commands and compare that to the overall runtime of tidy. If it's the longest chain then parallelizing makes sense.

Reply:
#143032 happens to have some numbers. We should probably move some of the longer-running tests to the front of main so that other ones get a chance to execute in parallel with them.